siliconcompiler/schema_support/pathschema.py

```
@@ -358,7 +375,9 @@ def active_dataroot(self, dataroot: str = None):
             Sets the file to top.v and associates lambdalib as the dataroot.
         '''
 
-        if dataroot and dataroot not in self.getkeys("dataroot"):
+        schema = self.__dataroot_section()
+
+        if dataroot and dataroot not in schema.getkeys("dataroot"):
             raise ValueError(f"{dataroot} is not a recognized dataroot")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)